xinference/api/restful_api.py

```diff
@@ -838,6 +840,12 @@ async def launch_model(
                     detail="Invalid input. Allocated gpu must be a multiple of replica.",
                 )
 
+        if model_path is not None:
+            if not os.path.exists(model_path):
+                raise ValueError(
+                    f"Invalid input. `model_path`: {model_path} File or directory does not exist."
+                )
+
         if peft_model_config is not None:
```

Comment:
Remove this logic to `worker.py`. When this is a distributed launch case, users can specify `model_path` and `worker_ip` together to launch distributed custom model.